darts/dataprocessing/dtw/cost_matrix.py

```diff
@@ -0,0 +1,146 @@
+from typing import Tuple, Dict
+import numpy as np
+
+from .window import Window, CRWindow
+from abc import ABC, abstractmethod
+import array
+from itertools import repeat
+
+Elem = Tuple[int, int]
+
+
+class CostMatrix(ABC):
+    """
+    (n+1) x (m+1) Matrix
+    Cell (i,j) corresponds to minimum total cost/distance of matching elements (i-1, j-1) in series1 and series2.
+
+    Row 0 and column 0, are typically set to infinity, to prevent matching before the first element
+    """
+
+    n: int
+    m: int
+
+    @abstractmethod
+    def fill(self, value: np.float): pass
+
+    @abstractmethod
+    def __getitem__(self, item): pass
+
+    @abstractmethod
+    def __setitem__(self, key, value): pass
+
+    @abstractmethod
+    def __iter__(self): pass
+
+    @abstractmethod
+    def to_dense(self) -> np.ndarray:
+        """
+
+        Returns
+        -------
+        Dense n x m numpy array, where empty cells are set to np.inf
+        """
+        pass
+
+    @staticmethod
+    def from_window(window: Window):
+        """
+        Creates a cost matrix from a window function.
+        Depending on the density of the active cells in the window,
+        will select either a dense or sparse storage representation.
+
+        Parameters
+        ----------
+        window: `Window`
```

Comment:
Add a description below `window`. Mention the type of window in the description and remove it from this line